### PR TITLE
Add contributing page to the docs (#1197)

### LINKED ISSUE #1197: [DOCS] Contribution docs page
Similar to https://www.python-httpx.org/contributing/ but addapted to uvicorn.

### PR DESCRIPTION
**Pull request by humrochagf**
A small change to the testing script was added to follow the same testing capabilities from other encode projects, like the ability to run a single testing file or test during development.

EDIT by @Kludex : Closes #1197

### CODE REVIEW

**Comment by humrochagf** [comment]
Just the releasing part that I'm not quite sure it's 100% accurate

File: docs/contributing.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+Before releasing a new version, create a pull request that includes:
+
+- **An update to the changelog**:
+    - We follow the format from [keepachangelog](https://keepachangelog.com/en/1.0.0/).
+    - [Compare](https://github.com/encode/uvicorn/compare/) `master` with the tag of the latest release, and list all entries that are of interest to our users:
+        - Things that **must** go in the changelog: added, changed, deprecated or removed features, and bug fixes.
+        - Things that **should not** go in the changelog: changes to documentation, tests or tooling.
+        - Try sorting entries in descending order of impact / importance.
+        - Keep it concise and to-the-point. 🎯
+- **A version bump**: see `__version__.py`.
```
Comment (Kludex) [inline docs/contributing.md]:
```suggestion
- **A version bump**: see `__init__.py`.
```

File: docs/contributing.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
+# Contributing
+
+Thank you for being interested in contributing to Uvicorn.
+There are many ways you can contribute to the project:
+
+- Using Uvicorn on your stack and [reporting bugs/issues you find](https://github.com/encode/uvicorn/issues/new)
+- [Implementing new features and fixing bugs](https://github.com/encode/uvicorn/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22)
```
Comment (Kludex) [inline docs/contributing.md]: We actually don't use the "good first issue" label, maybe "help wanted"?
```suggestion
- [Implementing new features and fixing bugs](https://github.com/encode/uvicorn/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22)
```

File: docs/contributing.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+- Code snippet
+- Error traceback
+
+You should always try to reduce any examples to the *simplest possible case*
+that demonstrates the issue.
+
+Some possibly useful tips for narrowing down potential issues...
+
+- Does the issue exist with a specific supervisor like `Multiprocess` or more than one?
+
```
Comment (Kludex) [inline docs/contributing.md]:
```suggestion
```

File: docs/contributing.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+- Does the issue exist on asgi, or wsgi, or both?
+- Does the issue exist with `Client`, `AsyncClient`, or both?
```
Comment (Kludex) [inline docs/contributing.md]:
```suggestion
```

**Review by Kludex — CHANGES_REQUESTED** [review]
Thanks for the PR @humrochagf ! :tada:

**Comment by humrochagf** [comment]
Is there anything missing for this one?

**Review by euri10 — APPROVED** [review]
looks good to me, I let @Kludex push the final go !

**Review by Kludex — APPROVED** [review]
Thanks again @humrochagf ! :smile:

**Pull request merged**